powertools-core/src/main/java/software/amazon/lambda/powertools/core/internal/LambdaHandlerProcessor.java

```diff
@@ -22,7 +22,7 @@
 import org.aspectj.lang.ProceedingJoinPoint;
 
 public final class LambdaHandlerProcessor {
-    private static String SERVICE_NAME = null != System.getenv("POWERTOOLS_SERVICE_NAME")
+    private static String SERVICE_NAME = System.getenv("POWERTOOLS_SERVICE_NAME") != null
```

Comment:
In order to keep it consistent with how we will like to check for null, I will prefer if we keep constant(null) to the left.

Reply:
Ok sure, i can revert this, just confusing to read left to right.

Maybe this would be cleaner and work with java8

```java
private static String SERVICE_NAME = Optional.ofNullable(System.getenv("POWERTOOLS_SERVICE_NAME")).orElse("service_undefined")
```

Reply:
@pankajagrawal16 ☝️

Reply:
@pankajagrawal16 `SERVICE_NAME` and `IS_COLD_START` look like constants, but are actually mutable due to the unit tests requiring this to be mutable. Maybe we should change either the unit tests or these misleading names.

Reply:
> Ok sure, i can revert this, just confusing to read left to right.
> 
> 
> 
> Maybe this would be cleaner and work with java8
> 
> 
> 
> ```java
> 
> private static String SERVICE_NAME = Optional.ofNullable(System.getenv("POWERTOOLS_SERVICE_NAME")).orElse("service_undefined")
> 
> ```

We have intentionally avoided use of optional here considering its a simple null check which belongs to an internal class.

I am a big fan of functional interfaces in java myself and have to always find a balance when I would like to use it.

Prefer using optional on public API that is public to let users ability to write functional code and not worry about nulls.

But these are just personal preferences too. Considering this being a internal class, I will still be happy to leave it the way it was. Thanks for the input. 🙂

Reply:
@pankajagrawal16  - ok i reverted back to the original code.